stock_picking_product_generate_sequence/models/stock_move_line.py

```diff
@@ -0,0 +1,49 @@
+##############################################################################
+# Copyright (c) 2022 brain-tec AG (https://braintec.com)
+# All Right Reserved
+#
+# See LICENSE file for full licensing details.
+##############################################################################
+from re import findall as regex_findall
+from re import split as regex_split
+
+from odoo import models, fields, api
+
+
+class StockMoveLine(models.Model):
+    _inherit = 'stock.move.line'
+
+    def action_generate_product_serial(self):
+        self.ensure_one()
+        self.lot_id = self.env['stock.production.lot'].create({
+            'product_id': self.product_id.id,
+            'company_id': self.company_id.id,
+            'name': self._get_next_serial(self.company_id, self.product_id) or
+                    self.env['ir.sequence'].next_by_code('stock.picking.product')
+        })
+
+    @api.model
+    def _get_next_serial(self, company, product):
+        sequence = product.product_tmpl_id.serial_lot
+        if sequence:
+            splitted = regex_split(r'(\d+)', sequence)
+            prefix = splitted[0]
+            initial_number = int(splitted[1])
+            padding = len(splitted[1])
+            last_serial = self.env['stock.production.lot'].search(
+                [
+                    ('company_id', '=', company.id),
+                    ('product_id', '=', product.id),
+                    ('name', '=ilike', '%s%%' % prefix)
+                ],
+                limit=1,
+                order='id DESC'
+            )
+
+            if last_serial:
+                last_serial_splitted = regex_split(r'(\d+)', last_serial.name)
+                last_serial_prefix = last_serial_splitted[0]
+                last_serial_number = int(last_serial_splitted[1])
+                return last_serial_prefix + str(last_serial_number + 1).zfill(len(last_serial_splitted[0]))
```

Comment:
[Minor] Better concatenate strings with format function than with +

Reply:
ok, I changed to join method

Reply:
Can you give me a real example for a `sequence` (`product.product_tmpl_id.serial_lot`) and also for a `last_serial.name`?
I want to check the return with a real example.

Reply:
an example of sequence creation with the pattern defined in the product.
product.product_tmpl_id.serial_lot: 'TEST/00000'
last_serial.name: 'TEST/00007'

Reply:
ok, thanks for the exmaples.
Just to ask. Is the format of the `name` in the `stock.production.lot` also guaranteed so that we get `last_serial_prefix` and `last_serial_number`?

I don't know if it's possible to change the `name` in the `stock.production.lot` manually in the GUI or with an import, or if the ir.sequence is used and has a different format configured than we expect.

Reply:
yes, it is possible to change the `name` in `stock.production.lot` from the GUI. I have tested it and the field is editable.

Reply:
My question was more, if it's possible to edit the `stock.production.lot` and write e.g. `ABC` as `name`.
In that case, maybe your code could crash. 
But I'm not sure, if this is possible. I suggest that you quickly try to use `ABC` as name and let your code run.
Thanks :smile:

Reply:
As you say, I tried to change the `name` of the `stock.production.lot` I generated and there was no problem. You can put anything in the `name` and it will appear in the line without any problem.
Here is a screenshot
![image](https://user-images.githubusercontent.com/97018161/186633015-3337fdce-043d-4eb7-83a5-98af4528c2b4.png)

Reply:
I could not test it because I don't have a DB from SAH and I could not configure it correctly that I could test.
But I believe you.
I would check here if `last_serial_splitted` really has 2 elements, because we refer directly to `last_serial_splitted[0]` and `last_serial_splitted[1]`, so I want to be sure there are such elements.

Reply:
Thank you for believing in me, I attached a screenshot to show the values when debugging
![sah_last_serial](https://user-images.githubusercontent.com/97018161/187166482-74656a8f-3908-4c12-a501-0d82e0df32b1.png)
But I understand, and It is important to check if the values exist. I added the condition to check if both values exist. thx